client/src/document/index.test.tsx

```diff
@@ -1,11 +1,36 @@
 import { Route, Routes, MemoryRouter } from "react-router-dom";
-import React from "react";
 import { render, waitFor } from "@testing-library/react";
 
 import { Document } from "./index";
 
 declare var global: Window;
 
+// TODO: move this somewhere better once we have more use cases.
+class IntersectionObserver {
```

Comment:
Considering our chat about progressive enhancement the other day, might it be better to remove this from the tests, and just ensure we wrap our code with `if (typeof IntersectionObserver !== "undefined")` or similar, as necessary?